platform_umbrella/apps/common_core/lib/common_core/state_summary/postgres_state.ex

```diff
@@ -45,4 +45,24 @@ defmodule CommonCore.StateSummary.PostgresState do
   def user_secret(_state_summary, %Cluster{name: cluster_name} = _cluster, %PGUser{username: username} = _user) do
     Enum.join(["cloudnative-pg", "pg-" <> cluster_name, username], ".")
   end
+
+  @spec password_for_user(
+          CommonCore.StateSummary.t() | any(),
+          CommonCore.Postgres.Cluster.t() | nil,
+          CommonCore.Postgres.PGUser.t() | nil
+        ) :: binary()
+  def password_for_user(_state_summary, nil = _cluster, nil = _user), do: ""
+  def password_for_user(_state_summary, nil = _cluster, _user), do: ""
+  def password_for_user(_state_summary, _cluster, nil = _user), do: ""
+
+  def password_for_user(
+        _state_summary,
+        %Cluster{password_versions: versions} = _cluster,
+        %PGUser{username: username} = _user
+      ) do
+    versions
+    |> Enum.sort_by(& &1.version, :desc)
+    |> Enum.find(%{password: ""}, &(&1.username == username))
+    |> Map.get(:password, "")
```

Comment:
Why empty string rather than nil as the return value when there's no password found?

Reply:
No particular reason. It's easy enough to change to nil so I'll do that right quick.